src/pages/signin/SignInPage.js

```diff
@@ -85,7 +88,18 @@ class SignInPage extends Component {
         // - AND an account did not exist or is not validated for that login
         const showResendValidationLinkForm = this.props.credentials.login && !validAccount;
 
-        const welcomeText = this.props.translate(`welcomeText.${showPasswordForm ? 'phrase4' : 'phrase1'}`);
+        const welcomeText = showResendValidationLinkForm && this.props.account.validateCodeExpired
+            ? ''
+            : this.props.translate(`welcomeText.${showPasswordForm ? 'phrase4' : 'phrase1'}`);
+
+        const validateCodeExpired = lodashGet(this.props.account, 'validateCodeExpired', false);
+
+        const resendLinkTitleMessage = this.props.translate(`resendValidationForm.${validateCodeExpired ? 'validationCodeFailedMessage' : 'weSentYouMagicSignInLink'}`,
```

Comment:
i think there is too much here. This line is not very readable and it took me a while to understand why loginType is there for `validationCodeFailedMessage` (took me a while to realize its not needed for validationCodeFailedMessage but is used just for weSentYouMagicSignInLink)

I personally feel this would be easier read if written like so,
```jsx
let resendLinkTitleMessage = '';
if (validateCodeExpired) {
    resendLinkTitleMessage = this.props.translate('resendValidationForm.validationCodeFailedMessage');
} else if (showResendValidationLinkForm) {
    resendLinkTitleMessage = this.props.translate('resendValidationForm.weSentYouMagicSignInLink', {
        loginType: (Str.isSMSLogin(this.props.credentials.login)
            ? this.props.translate('common.phoneNumber').toLowerCase()
            : this.props.translate('common.email')).toLowerCase(),
    });
}
```

Reply:
Okay, thanks. I'll update and raise the PR. Somewhere earlier I had seen that we prefer ternary operator over `if-else` block.

Reply:
Yeah, normally we prefer ternary. This situation is a little different though, with the ternary the operations and arguments involved become hard to follow so I feel like if-else is a lot cleaner here.